src/android/BluetoothLePlugin.java

```diff
@@ -2905,6 +2905,9 @@ public void onScanResult(int callbackType, ScanResult result) {
           JSONObject returnObj = new JSONObject();
 
           addDevice(returnObj, result.getDevice());
+          if(result.getScanRecord().getDeviceName() != null){
+            addProperty(returnObj, keyName, result.getScanRecord().getDeviceName().replace("\0", ""));
```

Comment:
I think that the behaviour of the plugin should not change, so since getDevice().getName() skips the zeroes, I think the result of getScanRecord().getDeviceName() should be filtered to obtain the same result.